app/models/user.rb

```diff
@@ -17,11 +17,25 @@ def next_slot_in_queue
   end
 
   def remove_from_queue!(queue_item)
-    if queue_items.include? queue_item
-      queue_items.each do |i|
-        i.update_attribute :position, i.position - 1 if i.position > queue_item.position
+    queue_item.destroy if queue_items.include? queue_item
+  end
+
+  def has_in_queue?(video)
+    true if self.queue_items.map(&:video).include? video
```

Comment:
A more performant way to do this would be:

```ruby
self.queue_items.pluck(:video_id).include? video.id
```

`pluck` grabs the values, but doesn't need to instantiate every object in memory to do it. It grabs them straight from the database. Then you check if the video id exists in the plucked ids.

Also, `true if` is implicit, you don't need that since `include?` returns a boolean. You are saying "true if true".